src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleEqualityTests.cs

```diff
@@ -4050,9 +4230,182 @@ public static void Main()
                 Diagnostic(ErrorCode.ERR_ExpressionTreeContainsTupleBinOp, "t != t").WithLocation(9, 57)
                 );
         }
+
+        [Fact]
+        public void TestComparisonOfDynamicAndTuple()
+        {
+            var source = @"
+public class C
+{
+    (long, string) _t;
+    public C(long l, string s) { _t = (l, s); }
+
+    public static void Main()
+    {
+        dynamic d = new C(1, ""hello"");
+        (long, string) tuple1 = (1, ""hello"");
+        (long, string) tuple2 = (2, ""world"");
+        System.Console.Write($""{d == tuple1} {d != tuple1} {d == tuple2} {d != tuple2}"");
+    }
+    public static bool operator==(C c, (long, string) t)
+    {
+        return c._t.Item1 == t.Item1 && c._t.Item2 == t.Item2;
     }
+    public static bool operator!=(C c, (long, string) t)
+    {
+        return !(c == t);
+    }
+    public override bool Equals(object o) => throw null;
+    public override int GetHashCode() => throw null;
 }
+";
 
-// PROTOTYPE(tuple-equality)
-// Test with tuple element names (semantic model)
-// Test tuples with casts or nested casts
+            var comp = CreateCompilation(source, references: new[] { CSharpRef, SystemCoreRef }, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "True False False True");
```

Comment:
Is this testing pre-7.3 behavior?

Reply:
Yes. The behavior hasn't changed.

---
In reply to: [173848204](https://github.com/dotnet/roslyn/pull/25374#discussion_r173848204) [](ancestors = 173848204)